vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -136,6 +136,8 @@ def build(self):
             ),
             color="grey",
             parent_className="loading-container",
+            delay_show=500,
+            overlay_style={"visibility": "visible", "filter": "blur(2px)"},
```

Comment:
```suggestion
```

The `overlay_style` is such a nice feature! Amazing job @AnnMarieW 🚀 

I would have loved to implement a similar example as in your PR for the loading spinner, providing some opacity on the figure and having the loading spinner overlayed on top. This looks so nice visually! 🎨 

However, I just realized that this is not possible with Vizro, as we always always output an empty figure before the actual figure is loaded. So, the loading spinner is always displayed on top of an empty background. So we currently can't leverage `overlay_style` as I would like to. This might change in the future, in which case I would definitely go for some opacity on the figure 👍 

<img width="619" alt="Screenshot 2024-05-21 at 10 25 07" src="https://github.com/mckinsey/vizro/assets/90609403/1dfe16e4-2527-443f-a31e-b93d555e3056">

Reply:
I think this might already be useful for us actually once the figure has already loaded and you apply a slow filter/parameter 🙂

Reply:
Ah, yes, you are right! I tried out the timeout as you suggested; otherwise, it's not visible. In this case, @AnnMarieW, I would change it to below—that suffices for us! 💯 

```suggestion
       overlay_style={"visibility": "visible", "opacity": 0.3},
```